contracts/Paper.sol

```diff
@@ -68,10 +105,15 @@ contract Paper {
             approved += 1;
         else if (_state == State.Rejected)
             rejected += 1;
-        if (approved >= reviewers.length) {
-            paperState = State.Approved;
-        } else if (rejected >= reviewers.length / 2) {
-            paperState = State.Rejected;
-        }
+    }
+
+    function addFinalPaperState(State _state) public  primaryCheckedHasPassed() onlyOwner() {
+        require (paperState == State.onReview, "The final paper state has already been decided.");
+        uint index;
+        uint approved;
+        uint rejected;
+        (index, approved, rejected) = _getPaperData(msg.sender);
+        require(approved + rejected == reviewers.length, "Please wait until all reviewers have finished their review.");
+        paperState = _state;
```

Comment:
What does this function do ? Is it for the editor to choose whether to publish the validated paper in his journal ?

If so, I think it should be in the `PaperManager` contract, which is the contract that represents a journal containing multiple published papers, rather than in the `Paper` contract, which only represents the peer review process of a paper.